lms/services/grouping.py

```diff
@@ -0,0 +1,67 @@
+from lms.models import CanvasSection, Grouping
+from lms.models._hashed_id import hashed_id
+
+
+class GroupingService:
+    def __init__(self, db, application_instance_service, course_service):
+        self._db = db
+        self._application_instance = application_instance_service.get()
+        self._course_service = course_service
+
+    def upsert(self, grouping):
```

Comment:
Courses don't use this upsert yet as they have the added complexity of having to tables.